src/background/providers/cloudflare.test.ts

```diff
@@ -0,0 +1,334 @@
+            method: 'GET',
+        };
+
+        test('valid response with tokens', () => {
+            // @ts-ignore
+            const storage: Storage = new StorageMock();
+            const updateIcon = jest.fn();
+            const navigateUrl = jest.fn();
+
+            const provider = new CloudflareProvider(storage, { updateIcon, navigateUrl });
+            const tokens = [new Token(), new Token(), new Token()];
+            // @ts-ignore
+            provider.setStoredTokens(tokens);
+            const details = validDetails;
+            const result = provider.handleHeadersReceived(details);
+            expect(result).toEqual({ redirectUrl: details.url });
+            // Expect redeemInfo to be set.
+            // @ts-ignore
+            const redeemInfo = provider.redeemInfo;
+            expect(redeemInfo!.requestId).toEqual(details.requestId);
+            expect(redeemInfo!.token.toString()).toEqual(tokens[0].toString());
+            // Expect a token is used.
+            // @ts-ignore
+            const storedTokens = provider.getStoredTokens();
+            expect(storedTokens.map((token) => token.toString())).toEqual(
+                tokens.slice(1).map((token) => token.toString()),
+            );
+        });
+
+        test('valid response without tokens', () => {
+            // @ts-ignore
+            const storage: Storage = new StorageMock();
+            const updateIcon = jest.fn();
+            const navigateUrl = jest.fn();
+
+            const provider = new CloudflareProvider(storage, { updateIcon, navigateUrl });
+            // @ts-ignore
+            provider.setStoredTokens([]);
+            const details = validDetails;
+            const result = provider.handleHeadersReceived(details);
+            expect(result).toBe(undefined);
+        });
+
+        test('captcha.website response', () => {
+            // @ts-ignore
+            const storage: Storage = new StorageMock();
+            const updateIcon = jest.fn();
+            const navigateUrl = jest.fn();
+
+            const provider = new CloudflareProvider(storage, { updateIcon, navigateUrl });
+            const tokens = [new Token(), new Token(), new Token()];
+            // @ts-ignore
+            provider.setStoredTokens(tokens);
+            const details = validDetails;
+            details.url = 'https://captcha.website/';
+            const result = provider.handleHeadersReceived(details);
+            expect(result).toEqual(undefined);
+        });
+
+        test('invalid response', () => {
+            // @ts-ignore
+            const storage: Storage = new StorageMock();
+            const updateIcon = jest.fn();
+            const navigateUrl = jest.fn();
+
+            const provider = new CloudflareProvider(storage, { updateIcon, navigateUrl });
+            const tokens = [new Token(), new Token(), new Token()];
+            // @ts-ignore
+            provider.setStoredTokens(tokens);
+            const details = {
```

Comment:
just add a comment that tells what makes the call to `handleHeadersReceived` to be invalid, there are many reasons why `handleHeadersReceived` returns nothing.